tests/neg/LiquidClass1.hs

```diff
@@ -0,0 +1,39 @@
+module LiquidClass where
+
+
+-- | Typing classes
+-- | Step 1: Refine type dictionaries:
+
+class Compare a where
+	cmax :: a -> a -> a
+	cmin :: a -> a -> a
+
+instance Compare Int where	
+{-@ instance Compare Int where 
+	cmax :: Odd -> Odd -> Odd ;
+	cmin :: Int -> Int -> Odd
+  @-}
+
+	cmax y x = if x >= y then x else y
+  	cmin y x = if x >= y then x else y
+
+-- | creates dictionary environment:
+-- | * add the following environment
+-- | dictionary $fCompareInt :: Compare Int
+-- |               { $ccmax :: Odd -> Odd -> Odd
+-- |               , $ccmin :: Int -> Int -> Int
+-- |               }
+-- |
+-- | Important: do not type dictionaries, as preconditions
```

Comment:
Interesting, because now I just ignore that code, and nothing seems to break....

Have you tried in the original type classes to give in class methods preconditions?